enhancements/cluster-scope-secret-volumes/shared-resources-via-openshift-builds-operator.md

```diff
@@ -0,0 +1,291 @@
+- Removal of `CSIDriverSharedResource` feature for OCP 4.16
+
+
+### Upgrade / Downgrade Strategy
+
+No concerns for upgrade/downgrade, as `CSIDriverSharedResource` is a tech preview feature and does
+not support upgrades.
+
+
+### Version Skew Strategy
+
+OpenShift Builds operator will be responsible for managing upgrades of the csi driver.
+
+
+### Operational Aspects of API Extensions
+
+API extensions are currently managed by OLM and the operator. Webhooks related to these
+custom resources are the responsibility of the respective operator or operand, and generally should
+not validate pod admission.
+
+The exception is the pod admission webhook for the Shared Resource CSI driver, which has been
+discussed previously (see [shared resource validation](enhancements/cluster-scope-secret-volumes/shared-resource-validation.md)).
+
+
+#### Failure Modes
+
+Out of scope - will be described by the OpenShift Builds operator design.
+
+
+#### Support Procedures
+
+Out of scope - will be described by the OpenShift Builds operator documentation.
+
+
+## Implementation History
+
+TBD
+
+
+## Alternatives
+
+
+### Remain in OCP
+
+The CSI driver could remain in OpenShift, but would require significant changes to its validating
+admission webhook. Validating admission webhooks proved challenging to deploy with hosted control
+planes ("hypershift") when deployed in the OCP payload. Support for these webhooks exists today for
+OLM operators, as these run in the "guest" cluster and are not part of the hosted control plane.
+Future enhancements may allow OLM operators to generally deploy webhooks onto the hosted control plane.
+
+Replacing the current validating webhook with
+[CRD CEL expressions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules)
+(available in OCP 4.12) would prove to be challenging for the following reasons:
+
+- The webhook validates pod admission to check that `readOnly: true` is set on the referenced
+  volume. This is mainly to improve user experience, and could be removed due to scalability
+  concerns.
+- The webhook ensures that Secrets and ConfigMaps in any `openshift-*` namespace are not shared,
+  with the exception of those explicitly allowed by the driver webhook's
+  [configuration](https://github.com/openshift/csi-driver-shared-resource-operator/blob/master/assets/webhook/deployment.yaml#L37).
+
+
+### Deploy Shared Resources as a Separate Operator
+
+Rather than bundle the Shared Resources CSI driver with OpenShift Builds, the driver could be
+deployed with its own OLM operator. The primary issue with this approach is that OpenShift Builds
+would still want to deploy and manage this driver, since this is part of our strategy to make use
+of entiltements in builds simpler. This would likely add a lot of unnecessary complexity:
+
+- An additional operator CRD to reconcile (ex: `SharedResourceConfig`)
+- Version skews when OLM does its API resolution, OR
```

Comment:
version skews between what and what?

Reply:
Updated to reflect the concern with OLM dependency resolution